File: randcs-infrastructure/src/main/java/org/example/global/exception/ExceptionResponse.java
package org.example.global.exception;

import org.example.common.error.GlobalErrorCode;

import java.time.LocalDateTime;

public record ExceptionResponse(
        int statusCode,
        String message,
        String description,
        LocalDateTime timeStamp)
{
    public static ExceptionResponse of (GlobalErrorCode errorCode, String description) {
        return new ExceptionResponse(errorCode.getErrorCode(), errorCode.getErrorMessage(), description, LocalDateTime.now());
    }

    public static ExceptionResponse of (GlobalErrorCode errorCode) {
        return new ExceptionResponse(errorCode.getErrorCode(), errorCode.getErrorMessage(), errorCode.getErrorMessage(), LocalDateTime.now());
    }
}

File: randcs-infrastructure/src/main/kotlin/org/example/global/exception/ExceptionResponse.kt
package org.example.global.exception

import org.example.common.error.GlobalErrorCode
import java.time.LocalDateTime

data class ExceptionResponse(
    val statusCode: Int,
    val message: String,
    val description: String,
    val timeStamp: LocalDateTime
) {

    companion object {
        fun of(errorCode: GlobalErrorCode, description: String): ExceptionResponse {
            return ExceptionResponse(errorCode.errorCode, errorCode.errorMessage, description, LocalDateTime.now())
        }

        fun of(errorCode: GlobalErrorCode): ExceptionResponse {
            return ExceptionResponse(errorCode.errorCode, errorCode.errorMessage, errorCode.errorMessage, LocalDateTime.now())
        }
    }
}
